adapters/repos/db/replication.go

```diff
@@ -414,3 +428,90 @@ func (i *Index) IncomingDigestObjects(ctx context.Context,
 ) (result []replica.RepairResponse, err error) {
 	return i.digestObjects(ctx, shardName, ids)
 }
+
+func (db *DB) FetchObject(ctx context.Context,
+	class, shardName string, id strfmt.UUID,
+) (objects.Replica, error) {
+	index := db.GetIndex(schema.ClassName(class))
+	return index.readRepairGetObject(ctx, shardName, id)
+}
+
+func (i *Index) readRepairGetObject(ctx context.Context,
+	shardName string, id strfmt.UUID,
+) (objects.Replica, error) {
+	shard, ok := i.Shards[shardName]
+	if !ok {
+		return objects.Replica{}, fmt.Errorf("shard %q does not exist locally", shardName)
+	}
+
+	obj, err := shard.objectByID(ctx, id, nil, additional.Properties{})
+	if err != nil {
+		return objects.Replica{}, fmt.Errorf("shard %q read repair get object: %w", shard.ID(), err)
+	}
+
+	if obj == nil {
+		deleted, err := shard.wasDeleted(ctx, id)
+		if err != nil {
+			return objects.Replica{}, err
+		}
+		return objects.Replica{
+			ID:      id,
+			Deleted: deleted,
+		}, nil
+	}
+
+	return objects.Replica{
+		Object: obj,
+		ID:     obj.ID(),
+	}, nil
+}
+
+func (db *DB) FetchObjects(ctx context.Context,
+	class, shardName string, ids []strfmt.UUID,
+) ([]objects.Replica, error) {
+	index := db.GetIndex(schema.ClassName(class))
+	return index.fetchObjects(ctx, shardName, ids)
+}
+
+func (i *Index) fetchObjects(ctx context.Context,
+	shardName string, ids []strfmt.UUID,
+) ([]objects.Replica, error) {
+	shard, ok := i.Shards[shardName]
+	if !ok {
+		return nil, fmt.Errorf("shard %q does not exist locally", shardName)
+	}
+
+	objs, err := shard.multiObjectByID(ctx, wrapIDsInMulti(ids))
+	if err != nil {
+		return nil, fmt.Errorf("shard %q read repair multi get objects: %w", shard.ID(), err)
+	}
+
+	resp := make([]objects.Replica, len(ids))
+
+	for j, obj := range objs {
+		if obj == nil {
+			deleted, err := shard.wasDeleted(ctx, ids[j])
+			if err != nil {
+				i.logger.Errorf("failed to query object id %q deletion, skipping", ids[j])
+				continue
```

Comment:
At this point it is safe to just an error ( `return nil, error`)
FetchObjects() like MultiGetbjects() depend on objects being in a specific order (maybe changed in the future).

Reply:
fixed